src/rest_api/classes/person/widgets/lineage.clj

```diff
@@ -2,378 +2,298 @@
-
-(defn- query-this-person [person]
-  (let [db (d/entity-db person)
-        queried-data-this-person
-        (->> (d/q '[:find ?person-id ?person-name 
-                    :in $ ?person
-                    :where
-                    [?person :person/standard-name ?person-name]
-                    [?person :person/id ?person-id]]
-                  db (:db/id person))
-             (map (fn [oid]
-                    (generate-map-this-person oid)))
-             (flatten))]
-    (if queried-data-this-person queried-data-this-person)))
+    (for [value queried-data]
+          (if (or (nil? value)
+                  (nil? ((keyword (:other-person-id value)) scaling-map)))
+            1
+            (parse-int ((keyword (:other-person-id value)) scaling-map))))))
+
+(defn- person-node [direct-or-full person scaling-map]
+  {:id (str direct-or-full (:person-id person))
+   :name (:person/standard-name person)
+   :url (:person/id person)
+   :scaling (if-let [scaling ((keyword (:person/id person)) scaling-map)]
+              (parse-int scaling)
+              1)
+   :radius 100
+   :nodeshape "rectangle"})
+
+(defn- get-node [direct-or-full person largest-node-scaling scaling-map]
+  (let [scaling (if-let [scaling ((keyword (:person/id person)) scaling-map)]
+                  scaling
+                  1)]
+      {:id (str direct-or-full (:person/id person))
+       :name (:person/standard-name person)
+       :url (:person/id person)
+       :scaling scaling
+;       :radius (+ 25 (* 50 (/ (Math/log scaling) (Math/log largest-node-scaling))))
+       :nodeshape "ellipse"}))
+
+
+(defn- edges-supervisees [direct-or-full person]
+  (some->> (:person.supervised-by/_person person)
+           (map (fn [holder]
+                  (generate-map
+                    holder
+                    (:person/_supervised-by holder))))
+           (map (fn [supervisees]
+                  {:source (str direct-or-full (:person/id person))
+                   :target (str direct-or-full (:id (:name supervisees)))
+                   :role (:level supervisees)
+                   :targetArrowShape "triangle"
+                   :lineStyle "solid"
+                   :lineColor (get role-colour (first (:level supervisees)))}))))
+
+(defn- edges-supervisors [direct-or-full person]
+  (some->> (:person/supervised-by person)
+           (map (fn [holder]
+                  (generate-map
+                    holder
+                    (:person.supervised-by/person holder))))
+           (map
+             (fn [supervisor]
+               {:source (str direct-or-full (:id (:name supervisor)))
+                :target (str direct-or-full (:person/id person))
+                :role (first (:level supervisor))
+                :targetArrowShape "triangle"
+                :lineStyle (if (= direct-or-full "Direct") "dashed" "solid")
+                :lineColor (get role-colour (first (:level supervisor)))}))))
+
```

Comment:
As we've discussed elsewhere, depending on the depth of this graph-like relationship, 
these mutally recursive function calls can/will cause a StackOverflow exception.

I think for these (matually) recursive functions you might want to look into usign `trampoline`.
The mutual recursion is between the outer `defn recursive-*` function and the anonymous function invoked within the map expression in each function.
I believe these functions could be tweaked to use [trampoline][1] to avoid the StackOverflow issue; that being said I'd have to experiment as not used it before.

[1]: https://clojuredocs.org/clojure.core/trampoline

Reply:
If the recursion level is being hit, consider using a datalog query with (recursive) rules, which may well produce a similar solution.

Reply:
I did not know about trampoline. I think this is a great idea.